core/server/master/src/main/java/alluxio/master/meta/DefaultMetaMaster.java

```diff
@@ -268,7 +268,7 @@ public BackupResponse backup(BackupPOptions options) throws IOException {
     String dir = options.hasTargetDirectory() ? options.getTargetDirectory()
         : ServerConfiguration.get(PropertyKey.MASTER_BACKUP_DIRECTORY);
     UnderFileSystem ufs = mUfs;
-    if (options.getLocalFileSystem() && !ufs.getUnderFSType().equals("local")) {
+    if ((options.getLocalFileSystem() && !ufs.getUnderFSType().equals("local")) || !options.hasTargetDirectory()) {
```

Comment:
Can the user specify a non-local destination for periodic backups with this condition?

Reply:
they can't specify a new UFS type. If the user provides a path, it will backup to that path on the root UFS (consistent with current behavior). Allowing a destination aside from local and root UFS would require more complexity (ex: providing an S3 path when Alluxio doesn't have the Access/Secret Key)